### ISSUE TITLE
Code suggestions list doesn't receive keyboard focus when the list is short

### ISSUE THREAD

**Issue by ussuri** [ajaxorg/ace#2136]
The code suggestions list works well with the keyboard when there are enough items in it to trigger a scroller:

![screen shot 2014-08-28 at 4 15 40 pm](https://cloud.githubusercontent.com/assets/5606182/4210190/b8e0f5bc-3872-11e4-8c46-681472d229ab.png)

Keyboard arrow keys move around the list and ENTER commits the selection. However, when the list is so short that there is no scroller, the focus stays in the editor, and the only way to select an item (and even simply dismiss the list) is using the mouse:

![screen shot 2014-08-28 at 4 16 18 pm](https://cloud.githubusercontent.com/assets/5606182/4210202/ddfab798-3872-11e4-877f-759c813d81d1.png)

Note the cursor visible in the second screenshot: the arrow keys move it around, without affecting the list.
